darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -69,6 +92,52 @@ def __init__(
             f"{regression_model.lags}",
         )
 
+        raise_if(
+            regression_train_num_samples > 1 and not self._models_are_probabilistic(),
```

Comment:
Couldn't we accept a mix of probabilistic and deterministic models?
In the end we reduce the probabilistic base models anyway to a deterministic prediction so I think it's fair to use both kinds.
For all deterministic models we'd set num_samples dynamically to 1
WDYT?

Reply:
I agree with you, if reduction occurs, then the mix should be accepted